shared/lib/site-settings/exporter/reducers.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import Immutable from 'immutable';
+import debugModule from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TOGGLE_EXPORTER_ADVANCED_SETTINGS,
+	TOGGLE_EXPORTER_SECTION
+} from '../action-types';
+
+const debug = debugModule( 'calypso:exporter' );
+
+const initialUIState = Immutable.fromJS( {
+	advancedSettings: {
+		isVisible: false,
+		posts: {
+			isEnabled: true
+		},
+		pages: {
+			isEnabled: true
+		},
+		feedback: {
+			isEnabled: true
+		}
+	}
+} );
+
+/**
+ * Reducer for managing the exporter UI
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function ui( state = initialUIState, action ) {
+	switch ( action.type ) {
+		case TOGGLE_EXPORTER_SECTION:
+			debug( 'toggle section', action.section );
+			return state.updateIn( [ 'advancedSettings', action.section, 'isEnabled' ], ( x ) => ( !x ) );
+		case TOGGLE_EXPORTER_ADVANCED_SETTINGS:
+			debug( 'toggle advanced settings' );
+			return state.updateIn( [ 'advancedSettings', 'isVisible' ], ( x ) => ( !x ) );
+	}
+
+	return state;
+}
+
+export default combineReducers( {
+	ui
+} );
```

Comment:
I'm not sure how hidden this feature is, as it's [part of the official documentation](https://github.com/rackt/redux/blob/master/docs/api/combineReducers.md) on it. There's [nothing magic about it](https://github.com/rackt/redux/blob/master/src/utils/combineReducers.js#L117) either. If you know you will be using this for more subtrees, then it's fine to leave in, but if not then I'd recommend taking it out. It's not too bad to add it in as we go to avoid jumping ahead of ourselves, though this way doesn't look as pretty.

```js
export default ( state, action ) => ( { ui: ui( state, action ) } );
```